### ISSUE TITLE
[ENH] TimeLLM for forecasting

### ISSUE THREAD

**Issue by phoeenniixx** [sktime/sktime#7390]
Found some good paper, will they be a good addition?
TimeLLM  : [Paper](https://arxiv.org/abs/2310.01728) .  [Code](https://github.com/KimMeen/Time-LLM)

**Comment by fkiraly** [comment]
Interesting. Are these perhaps one one of two other lists?

* foundation models: https://github.com/sktime/sktime/issues/6177
* Fedformer: https://github.com/sktime/sktime/issues/4939

**Comment by phoeenniixx** [comment]
Yes I think Fedformer is in already in the issues although not so sure about TimeLLM.
TimeLLM is a rather recent paper submitted in Oct'23 and last revised in Jan'24, it uses LLMs like GPT2, Llama etc for forecasting, I haven't looked in all the issues https://github.com/sktime/sktime/issues/6177  but as far as i can see, TimeLLM is not  in the list and I think it will be a good addition to `sktime` under the pre-trained models/LLMs like some other models in #6177. 
TimeLLM makes use of Prompt embedding and Patch reprogramming to make LLMs to predict time series.

**Comment by fkiraly** [comment]
Yes, it looks TimesLLM is not contained in #4939.

**Comment by fkiraly** [comment]
I have also added TimeLLM to the umbrella issue: https://github.com/sktime/sktime/issues/6177

**Comment by julian-fong** [comment]
Looks like this one will be a bit tricky to interface, considering there is no official package or uploads to popular open-source modeling repositories like huggingface

**Comment by jgyasu** [comment]
I would like to be assigned to this issue. I'll have to do vendoring so I want to know if it would be impactful enough or not? If this isn't a priority for sktime right now then I'd like to work on interfacing any other foundational model and it will be helpful if you can point me to any @fkiraly

**Comment by fkiraly** [comment]
Yes, this one would be a very nice to have!

There is still no package for the original, so vendoring would still be the way to go. Vendor libraries are in `sktime.libs`.
